Calendar/Calendar.js

```diff
@@ -0,0 +1,318 @@
+// @flow
+import React, { Component } from 'react';
+import { View, Text, StyleSheet, TouchableOpacity, Image } from 'react-native';
+import moment from 'moment';
+import Back from './Back.png';
+
+type Props = {
+  onChange: (value: Date) => void,
+  value?: Date,
+  width?: number,
+  defaultView?: string,
+  selectedDateColor?: string,
+}
+
+type TogggleProps = {
+  onPress: () => void,
+}
+
+type HeaderProps = {
+  onPress: () => void,
+  value: string,
+}
+
+type State = {
+  activeView: string,
+  currentMonth: Date,
+  currentYear: Date,
+}
+
+const calPadding = 5;
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: 'white',
+    borderWidth: 1,
+    borderRadius: 5,
+    padding: calPadding,
+  },
+  headerTextStyle: { fontSize: 20, fontWeight: 'bold' },
+  headerStyle: { flexDirection: 'row' },
+  headerTextContainer: { paddingRight: 10 },
+  calRow: { flexDirection: 'row', justifyContent: 'space-between' },
+  centerItems: { alignItems: 'center', justifyContent: 'center' },
+  calendar: { flexDirection: 'row', flexWrap: 'wrap', justifyContent: 'space-between' },
+  isSelectedContainer: { borderRadius: 5 },
+  isSelectedText: { color: 'white' },
+});
+
+const Prev = ({ onPress }: TogggleProps) => (
+  <TouchableOpacity style={styles.centerItems} onPress={onPress}>
+    <Image source={Back} />
+  </TouchableOpacity>
+);
+
+const Next = ({ onPress }: TogggleProps) => (
+  <TouchableOpacity style={styles.centerItems} onPress={onPress}>
+    <Image source={Back} style={{ transform: [{ rotate: '180deg' }] }} />
+  </TouchableOpacity>
+);
+
+const Header = ({ value, onPress }: HeaderProps) => (
+  <TouchableOpacity onPress={onPress} style={styles.headerTextContainer}>
+    <Text style={styles.headerTextStyle}>
+      {value}
+    </Text>
+  </TouchableOpacity>
+);
+class Calendar extends Component<Props, State> {
+  static defaultProps = {
+    width: 300,
+    value: '',
+    defaultView: 'day',
+    selectedDateColor: '#ffd400',
+  };
+
+  state = {
+    activeView: this.props.defaultView,
+    currentDate: this.props.value 
+      ? moment(this.props.value).startOf(this.props.defaultView === 'year' ? 'year' : 'month')
```

Comment:
use constructor, it will remove repetition that is occurring here,

```javascript
  constructor(props) {
  super(props);
  const { value, defaultView } = props;
  const viewType = (defaultView === 'year') ? 'year' : 'month';
  const currentDate = value ? moment(value).startOf(viewType);
 this.state({
   activeView: defaultView,
   currentDate,
 });

}
```

Reply:
i did just that, but somehow it wont read the value of props